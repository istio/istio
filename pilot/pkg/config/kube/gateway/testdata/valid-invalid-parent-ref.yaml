apiVersion: gateway.networking.k8s.io/v1
kind: GatewayClass
metadata:
  name: istio
spec:
  controllerName: istio.io/gateway-controller
---
apiVersion: gateway.networking.k8s.io/v1
kind: Gateway
metadata:
  name: gateway
  namespace: istio-system
spec:
  addresses:
  - value: istio-ingressgateway
    type: Hostname
  gatewayClassName: istio
  listeners:
  - name: default
    hostname: "*.domain.example"
    port: 80
    protocol: HTTP
    allowedRoutes:
      namespaces:
        from: All
---
apiVersion: gateway.networking.k8s.io/v1
kind: HTTPRoute
metadata:
  name: mixed-parent-ref-validity
  namespace: default
spec:
  parentRefs:
  # valid refs
  - name: gateway
    namespace: istio-system
    port: 80
  - kind: ServiceEntry
    group: networking.istio.io
    name: egress
  # invalid refs
  - name: gateway
    namespace: istio-system
    port: 1234 # invalid port
  - group: ''
    kind: Service # service does not exist
    name: random
    namespace: nonexistent
  rules:
  - backendRefs:
    - name: httpbin
      port: 80
      weight: 1
---
apiVersion: networking.istio.io/v1
kind: ServiceEntry
metadata:
  name: egress
  namespace: default
spec:
  hosts:
  - "google.com"
  ports:
  - number: 80
    name: http
    protocol: HTTP
  - number: 443
    name: tls
    protocol: TLS