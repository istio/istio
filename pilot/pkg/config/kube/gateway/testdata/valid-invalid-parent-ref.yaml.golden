apiVersion: networking.istio.io/v1
kind: Gateway
metadata:
  annotations:
    internal.istio.io/gateway-semantics: gateway
    internal.istio.io/gateway-service: istio-ingressgateway.istio-system.svc.domain.suffix
    internal.istio.io/parents: Gateway/gateway/default.istio-system
  creationTimestamp: null
  name: gateway-istio-autogenerated-k8s-gateway-default
  namespace: istio-system
spec:
  servers:
  - hosts:
    - '*/*.domain.example'
    port:
      name: default
      number: 80
      protocol: HTTP
---
apiVersion: networking.istio.io/v1
kind: VirtualService
metadata:
  annotations:
    internal.istio.io/parents: HTTPRoute/mixed-parent-ref-validity.default
    internal.istio.io/route-semantics: gateway
  creationTimestamp: null
  name: default~google.com
  namespace: default
spec:
  gateways:
  - mesh
  hosts:
  - google.com
  http:
  - name: default.mixed-parent-ref-validity.0
    route:
    - destination:
        host: httpbin.default.svc.domain.suffix
        port:
          number: 80
---
apiVersion: networking.istio.io/v1
kind: VirtualService
metadata:
  annotations:
    internal.istio.io/parents: HTTPRoute/mixed-parent-ref-validity.default
    internal.istio.io/route-semantics: gateway
  creationTimestamp: null
  name: istio-system~gateway-istio-autogenerated-k8s-gateway-default~*
  namespace: default
spec:
  gateways:
  - istio-system/gateway-istio-autogenerated-k8s-gateway-default
  hosts:
  - '*'
  http:
  - name: default.mixed-parent-ref-validity.0
    route:
    - destination:
        host: httpbin.default.svc.domain.suffix
        port:
          number: 80
---
